### ISSUE TITLE
Python build hanging forever

### ISSUE THREAD

**Issue by gappleto97** [Shippable/support#2693]
* Build: https://app.shippable.com/runs/575a22694d84240c00e5c2fb
* Script: https://github.com/gappleto97/p2p-project/blob/beta/py/test.py

The relevant methods are `test_net_sans_network` and `test_net_connection`. I believe it's failing on the latter, but I have no easy way to know this since there's no output given.

Here's a link to the travis build of this which completely works: https://travis-ci.org/gappleto97/p2p-project/builds/136593878

You will also note that this works if you run it on your own system.

**Comment by gappleto97** [comment]
https://github.com/gappleto97/p2p-project/commit/583f5182b6034bc554218bee9f4705ecf8ca5189

This commit seemed to fix things. It would appear that your environment reacts poorly to `multiprocessing.pool.ThreadPool` being used. I'm not certain if this is an issue to file with you or CPython, though.

I can say for sure that this works on Travis-CI, as well as on my system. This makes me think it has something to do with how you're virtualizing systems.

**Comment by pranaypareek** [comment]
@gappleto97 we looked into the problem. The only thing I could zero down on right now is that this problem only occurs with Python 2.6

I didn't see this happen with other versions of python that I tested (the ones specified in shippable.yml)

I put up logs in the functions `test_net_sans_network` & `test_net_connection` to debug where the problem might be. Turns out both these functions tend to get stuck when functions from `net.py` file are called (which in turn makes use of `multiprocessing.pool`)

What's peculiar is that this works with other Python versions. So the problem might be due to the way `multiprocessing.pool` is interacting with the environment around Python 2.6

We'll investigate further and get back to you with our findings.
